src/NuGet.Core/NuGet.Build.Tasks.Console/MSBuildStaticGraphRestore.cs

```diff
@@ -892,7 +892,9 @@ private PackageSpec GetPackageSpec(IMSBuildProject project, IReadOnlyDictionary<
             restoreMetadata.RestoreLockProperties = new RestoreLockProperties(project.GetProperty("RestorePackagesWithLockFile"), project.GetProperty("NuGetLockFilePath"), project.IsPropertyTrue("RestoreLockedMode"));
             restoreMetadata.Sources = GetSources(project, innerNodes, settings);
             restoreMetadata.TargetFrameworks = GetProjectRestoreMetadataFrameworkInfos(targetFrameworkInfos, projectsByTargetFramework);
-
+            restoreMetadata.UsingMicrosoftNETSdk = MSBuildRestoreUtility.GetUsingMicrosoftNETSdk(project.GetProperty("UsingMicrosoftNETSdk"));
+            NuGetVersion skdAnalysisLevel = MSBuildRestoreUtility.GetSdkAnalysisLevel(project.GetProperty("SdkAnalysisLevel"));
+            restoreMetadata.SdkAnalysisLevel = skdAnalysisLevel ?? restoreMetadata.SdkAnalysisLevel;
```

Comment:
it was supposed to be 
```CS
if (sdkAnalysisLevel is not null)
    {
        restoreMetadata.SdkAnalysisLevel = sdkAnalysisLevel;
    }
```
which would only update the value of ` restoreMetadata.SdkAnalysisLevel` if `sdkAnalysisLevel` is not null. I wanted to have this in a shorter form that's why updated it to `restoreMetadata.SdkAnalysisLevel = skdAnalysisLevel ?? restoreMetadata.SdkAnalysisLevel;` which updates the value if `sdkAnalysisLevel` is not null otherwise leaves it as is.  Another way to go about it is to not do a null check and just assign the value: `restoreMetadata.SdkAnalysisLevel = skdAnalysisLevel`. However, in the case `sdkAnalysisLevel` is null, if `restoreMetadata.SdkAnlysisLevel` has a value, we would be overwriting it. I can't think of a scenario where this happens, but I was just being extra cautious.

Reply:
In my opinion, the `??` pattern is great if you have a logical default.  In this case, you really just don't want to set the property if a value was not given so I would just go with the expanded version using the `if` statement.